Update doc about implementation on sandbox (#12)
edgexfoundry/jenkins_pipeline_presentation: README.md

### Build final docker image

`docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`

## Deployment On Sandbox

The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**

#### Get Access To Sandbox

> The official process to request for access to sandbox is provided in the link **[Here](https://https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**

Reviewer: The link URL is broken due to having "https://https://" at the beginning.

Author: fixed

> If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.

Reviewer: I don't think this line should be here. The site linked above contains the needed info, and this is misleading. Yes, you'll use the same LFID to access the sandbox as prod, but this seems to imply that if you have access to prod, you immediately have sandbox access, which is not true. As the docs show, each user has to request access to the sandbox from LF.

Author: Updated.

#### Implementation

> Once you create a new Jenkinsfile follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. Since, the Jenkins sandbox will recognizes only the file Jenkinsfile.sandbox to execute.

Reviewer: ```suggestion
Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will only execute the Jenkinsfile.sandbox file.```

Author: Updated

#### Configurations on Jenkins UI

For the EdgeX Foundry Project Org all the configurations such as Credentials to access the github repos were already provided. Once you add Jenkinsfile.sandbox to the repository, Now you scan the organization on Jenkins UI to execute the jobs.

![edgexjenkinsorg.png](images/edgexjenkinsorg.PNG)

> It already has set the behaviour to discover the branches, pull request from origin and pull requests from forks.

Reviewer: ```suggestion
The Jenkins pipeline job will automatically discover branches and pull requests for the repository.
```

Author: Updated

### Build final docker image

`docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`

## Deployment On Sandbox

The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**

#### Get Access To Sandbox

> To deploy the pipeline on test environment, You need to request for access to sanbox through **[LF Service Desk](https://jira.linuxfoundation.org/servicedesk/customer/portals)**

Reviewer 2: The official process to request access is found [on the lf-releng docs site](https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html)

Reviewer 2: https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox

Reviewer 2: Once the LF ID exists, the request can be made to add via JIRA ticket request.

Author: I will update these changes mentioned.

> The official process to request access sandbox is provided in the link **[Here](https://https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**

Reviewer 3: change as 'access request'

Author: fixed

If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox.

#### Implementation

> Once you create a new jenkins file follow the procedure as above. In your github repository, your jenkinsfile name should be Jenkinsfile.sanbox. Since, the jenkins sanbox will recognizes only the file Jenkinsfile.sanbox to execute.

Reviewer 2: spelling eror - please change to correct spelling - "sandbox"

Reviewer 2: I think we wanted to get away from using a separate Jenkinsfile based on some initial testing.

Author: what i understood is we using the same jenkinsfile for box prod and sandbox but we just changing the name of the file right!

> Once you create a new Jenkins file follow the procedure as above. In your Github repository, your jenkinsfile name should be Jenkinsfile.sandbox. Since, the jenkins sanbox will recognizes only the file Jenkinsfile.sandbox to execute.

Reviewer 2: spelling - sandbox

Reviewer 2: capitalize Jenkins

Author: i apologize i donno  missed this some how

Author: fixed

> Once you create a new Jenkins file follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. Since, the Jenkins sandbox will recognizes only the file Jenkinsfile.sandbox to execute.

Reviewer 3: single word 'Jenkinsfile'

Author: fixed

#### Configurations on Jenkins UI

> For the EdgeX Foundry Project Org all the configurations such as Credentials to access the github repos were already provided. Once you add Jenkinsfile.sandbox to the repository, Now you scan the organization on jenkins UI to execute the jobs.

Reviewer 2: Jenkins

![edgexjenkinsorg.png](images/edgexjenkinsorg.PNG)

It already has set the behaviour to discover the branches, pull request from origin and pull requests from forks.

#### Things to remember

> * All the Jobs on jenkins Sanbox are automatically deleted every Saturday at 08:00 UTC

Reviewer 2: Jenkins - not jenkins

* Project configuration files and credentials are not loaded into the system

> * Jenkins nodes have the same OpenStack configuration as the production instance with    minor differences.

Reviewer 2: why mention the "minor differences"?

Author: fixed

### Build final docker image

`docker build -t docker-edgex-binary -f Dockerfile --build-arg BASE=base-build .`

## Deployment On Sandbox

The official site for Sandbox environment is **[https://jenkins.edgexfoundry.org/sandbox/](https://jenkins.edgexfoundry.org/sandbox/)**

#### Get Access To Sandbox

The official process to request for access to sandbox is provided in the link **[Here](https://lf-releng-docs.readthedocs.io/en/latest/jenkins-sandbox.html#get-access-to-the-jenkins-sandbox)**

If you already have the LFID and can access the Prod Environment you can use the same LFID to access the Sandbox, but before each other must request the access to sandbox by creating an LF ticket at **[LFServicedesk](https://jira.linuxfoundation.org/servicedesk/customer/portals)**

#### Implementation

> Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will execute only the file Jenkinsfile.sandbox file.

Reviewer: ```suggestion
Once you create a new Jenkinsfile, follow the procedure as above. In your Github repository, your Jenkinsfile name should be Jenkinsfile.sandbox. The Jenkins sandbox will execute only the Jenkinsfile.sandbox file.
```

Author: fixed it. Thanks
